Challenge2/src/agent.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  Finding,
+  FindingSeverity,
+  FindingType,
+  HandleTransaction,
+  TransactionEvent,
+  getEthersProvider,
+  ethers,
+} from "forta-agent";
+import { Provider } from "@ethersproject/providers";
+import { LRUCache } from "lru-cache";
+import { UNI_SWAP_EVENT_ABI, UNI_POOL_FUNCTIONS_ABI, UNI_FACTORY_ADDRESS, UNI_INIT_CODE_HASH } from "./constants";
+
+const addressIsUniCache = new LRUCache<string, boolean>({ max: 100000 });
+
+const getPoolValues = async (
+  provider: Provider,
+  uniPoolFunctionsAbi: string[],
+  poolAddress: string,
+  txEvent: TransactionEvent
+) => {
+  const poolContract = new ethers.Contract(poolAddress, uniPoolFunctionsAbi, provider);
+  const poolValues: any[] = await Promise.all([
```

Comment:
We really try to steer clear using `any` for types. Please update to be more specific.

Reply:
Not necessary, but a part of me wants `poolValues` to be an object rather than an array. That way it could be more readable to query their key. Having them in an array sort of treats them like they are similar items, but they're not. With `fee` especially being unlike the other two.

Again, not necessary, but something to keep in mind going forward. 🙂